Packages/com.latios.latiosframework/Mimic/Mecanim/Internal/MecanimInternalUtilities.cs

```diff
@@ -565,16 +579,22 @@ internal static bool Approximately(float a, in float b)
 
                     sampleTransform.position -= timeFragmentSample.position;
                     sampleTransform.rotation = math.mul(sampleTransform.rotation, math.inverse(timeFragmentSample.rotation));
+                    sampleTransform.scale /= timeFragmentSample.scale;
+                    sampleTransform.stretch /= timeFragmentSample.stretch;
                 }
 
                 previousRoot.position += sampleTransform.position * blendWeight;
                 previousRoot.rotation = math.slerp(previousRoot.rotation, math.mul(previousRoot.rotation, sampleTransform.rotation), blendWeight);
+                previousRoot.scale *= sampleTransform.scale * blendWeight;
+                previousRoot.stretch *= sampleTransform.stretch * blendWeight;
             }
 
             var rootDelta = TransformQvvs.identity;
 
             rootDelta.position = currentRoot.position - previousRoot.position;
             rootDelta.rotation = math.mul(currentRoot.rotation, math.inverse(previousRoot.rotation));
+            rootDelta.scale = currentRoot.scale / previousRoot.scale;
+            rootDelta.stretch = currentRoot.stretch / previousRoot.stretch;
```

Comment:
I think you have this backwards. currentRoot represents the root motion accumulated after the transition, whereas previousRoot represents the root motion accumulated before the hard exit time transition (rare). I believe we would want to add these, not subtract them.

Reply:
I don't think that's quite right.  Let's say the current root is 1.2f scale, and previous root is 1.2f scale.  If we multiplied them, that would make the result 1.44f, which would grow the transform instead of keeping it the same.  However, looking at the optimized skeleton system, I see I'm not multiplying the existing root by the delta, as I am in the exposed bone system.  I'll run a few more tests against both types of skeletons before I adjust it.